### DISCUSSION TITLE
Lantmäteriet har slutat fungera - januari 2022

### DISCUSSION THREAD

**Discussion by kiljano** [johanberonius/custom-map-source.appspot.com#6] [General]
Hej. Jag ville bara meddela att lantmäteriet min karta verkar ha slutat att fungera sedan en period. Tror du att man kan göra något åt det? Tack!

**Comment by johanberonius** [comment]
Lantmäteriet har tyvärr börjat blockera laddning av kartrutor från tredje part via https://minkarta.lantmateriet.se/  
Det gör att ingen av dessa kartor fungerar längre genom kartkällan i Guru Maps. Vilket är synd eftersom presentationen av kartan där är mycket bättre än i appen "Min Karta".

Tekniskt går det till så att alla HTTP-frågor som innehåller header-raden `User-Agent: GLMap` endast får svaret `400 Bad Request`. Detta går inte att komma runt genom att ändra i kartkällan. Om trafiken skulle gå via en proxy har den möjlighet att modifiera frågan på vägen.

Den gamla tjänsten KSO är numera nedstängd, så kartkällan som pekade på den fungerar inte heller längre.

Via Lantmäteriets API där en personlig API-nyckel krävs fungerar det fortfarande från Guru Maps. Men där är maximal zoom-nivå begränsad till 14. En onödigt strikt begränsning som gör kartan nästan oanvändbar.

**Comment by kiljano** [comment]
Jag ska pröva att skriva direkt till dom på lantmäteriet och fråga om dom
kan ändra detta. Med vänlig hälsning, Kiljan.

**Comment by johanberonius** [comment]
Kartkällorna för "Lantmäteriets Topografisk" och "Lantmäteriets Flygbild" är nu uppdaterade och bör fungera.
Övriga kartkällor från "Min karta" (terrängskuggning, historiska flygbilder och fastighetsgränser) är inte uppdaterade.

Lägg till i Guru Maps på https://custom-map-source.appspot.com/ (eller https://git.beroni.us/custom-map-source/)

**Reply by clindh** [reply]
Snyggt, hoppas den lösningen håller ett tag. Annars kanske Evgen kan tänka sig stoppa in stöd för User-Agent spoofing…

**Comment by kiljano** [comment]
Grymt! Jag skrev till dom häromdagen, jag vet inte om något sånt hjälper
men jag tycker att det är bra att dom får reda på när deras förändringar
förstör för användare, det är ju allmängods så att säga.
Tack ett bra arbete!!
